app/controllers/AmendCheckYourAnswersController.scala

```diff
@@ -52,16 +52,17 @@ class AmendCheckYourAnswersController @Inject() (
   def onPageLoad(): Action[AnyContent] = (identify andThen getData andThen requireData).async {
     implicit request =>
       val updatedAnswers = request.userAnswers.copy(changePage = None)
-      sessionRepository.clearChangePage(updatedAnswers) map { _ =>
-        navigator.firstMissingAnswer(updatedAnswers) match {
-          case Some(call) =>
-            // TODO - render "missing" version of CYA page
-            Redirect(call)
-          case None =>
+      navigator.firstMissingAnswer(updatedAnswers) match {
+        case Some(call) =>
+          // TODO - render "missing" version of CYA page
+          Future.successful(Redirect(call))
+        case None =>
+          sessionRepository.set(updatedAnswers) map { _ =>
```

Comment:
Fixed the implementation logic here to match main "create" CYA controller.  Only update the state of "changePage" in the db if there are no missing answers.